// gsc-typescript-sdk/dist/src/Model/Product/BusinessUnits.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BusinessUnits = void 0;
var BusinessUnits = /** @class */ (function () {
    function BusinessUnits() {
        this.collection = [];
    }
    BusinessUnits.prototype.findByOperatorCode = function (operatorCode) {
        var businessUnit = this.collection.find(function (_businessUnit) { return _businessUnit.getOperatorCode() === operatorCode; });
        if (!businessUnit)
            return null;
        return businessUnit;
    };
    BusinessUnits.prototype.searchByBusinessUnit = function (businessUnit) {
        var result = [];
        this.collection.forEach(function (_businessUnit) {
            if (_businessUnit.getBusinessUnit() == businessUnit) {
                result.push(_businessUnit);
            }
        });
        return result;
    };
    BusinessUnits.prototype.all = function () {
        return this.collection;
    };
    BusinessUnits.prototype.add = function (businessUnit) {
        this.collection.push(businessUnit);
    };
    return BusinessUnits;
}());
exports.BusinessUnits = BusinessUnits;
