// gsc-typescript-sdk/dist/src/Model/Product/GlobalProducts.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.GlobalProducts = void 0;
var GlobalProducts = /** @class */ (function () {
    function GlobalProducts() {
        this.collection = [];
    }
    GlobalProducts.prototype.findBySellerSku = function (sellerSku) {
        var product = this.collection.find(function (product) {
            return product.getSellerSku() === sellerSku;
        });
        if (!product)
            return null;
        return product;
    };
    GlobalProducts.prototype.searchByName = function (name) {
        var result = [];
        this.collection.forEach(function (product) {
            if (product.getName() == name)
                result.push(product);
        });
        return result;
    };
    GlobalProducts.prototype.all = function () {
        return this.collection;
    };
    GlobalProducts.prototype.add = function (product) {
        this.collection.push(product);
    };
    return GlobalProducts;
}());
exports.GlobalProducts = GlobalProducts;
