// gsc-typescript-sdk/dist/src/Model/Brand/Brands.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Brands = void 0;
var Brands = /** @class */ (function () {
    function Brands() {
        this.collection = [];
    }
    Brands.prototype.findById = function (brandId) {
        var brand = this.collection.find(function (brand) { return brand.getBrandId() == brandId; });
        if (!brand)
            return null;
        return brand;
    };
    Brands.prototype.searchByName = function (name) {
        var result = [];
        this.collection.forEach(function (brand) {
            if (brand.getName() == name)
                result.push(brand);
        });
        return result;
    };
    Brands.prototype.searchByGlobalIdentifier = function (globalIdentifier) {
        var result = [];
        this.collection.forEach(function (brand) {
            if (brand.getGlobalIdentifier() == globalIdentifier)
                result.push(brand);
        });
        return result;
    };
    Brands.prototype.all = function () {
        return this.collection;
    };
    Brands.prototype.add = function (brand) {
        this.collection.push(brand);
    };
    return Brands;
}());
exports.Brands = Brands;
