// gsc-typescript-sdk/dist/src/Contract/BusinessUnitOperatorCodes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.COUNTRY_OPERATOR = exports.COUNTRY_CODES = exports.OPERATOR_CODES = void 0;
var OPERATOR_CODES;
(function (OPERATOR_CODES) {
    OPERATOR_CODES["FALA_CHILE"] = "facl";
    OPERATOR_CODES["FALA_PERU"] = "fape";
    OPERATOR_CODES["FALA_MEXICO"] = "famx";
    OPERATOR_CODES["FALA_COLOMBIA"] = "faco";
    OPERATOR_CODES["FALA_ARGENTINA"] = "faar";
})(OPERATOR_CODES || (OPERATOR_CODES = {}));
exports.OPERATOR_CODES = OPERATOR_CODES;
var COUNTRY_CODES;
(function (COUNTRY_CODES) {
    COUNTRY_CODES["CODE_CHILE"] = "cl";
    COUNTRY_CODES["CODE_PERU"] = "pe";
    COUNTRY_CODES["CODE_MEXICO"] = "mx";
    COUNTRY_CODES["CODE_COLOMBIA"] = "co";
    COUNTRY_CODES["CODE_ARGENTINA"] = "ar";
})(COUNTRY_CODES || (COUNTRY_CODES = {}));
exports.COUNTRY_CODES = COUNTRY_CODES;
var COUNTRY_OPERATOR = {};
exports.COUNTRY_OPERATOR = COUNTRY_OPERATOR;
COUNTRY_OPERATOR[COUNTRY_CODES.CODE_CHILE] = OPERATOR_CODES.FALA_CHILE;
COUNTRY_OPERATOR[COUNTRY_CODES.CODE_PERU] = OPERATOR_CODES.FALA_PERU;
COUNTRY_OPERATOR[COUNTRY_CODES.CODE_MEXICO] = OPERATOR_CODES.FALA_MEXICO;
COUNTRY_OPERATOR[COUNTRY_CODES.CODE_COLOMBIA] = OPERATOR_CODES.FALA_COLOMBIA;
COUNTRY_OPERATOR[COUNTRY_CODES.CODE_ARGENTINA] = OPERATOR_CODES.FALA_ARGENTINA;
