// gsc-typescript-sdk/dist/src/Contract/ProductAttribute.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Attribute = void 0;
var Attribute;
(function (Attribute) {
    Attribute["FEED_SELLER_SKU"] = "SellerSku";
    Attribute["FEED_NEW_SELLER_SKU"] = "NewSellerSku";
    Attribute["FEED_NAME"] = "Name";
    Attribute["FEED_DESCRIPTION"] = "Description";
    Attribute["FEED_PRIMARY_CATEGORY"] = "PrimaryCategory";
    Attribute["FEED_PRICE"] = "Price";
    Attribute["FEED_BRAND"] = "Brand";
    Attribute["FEED_PRODUCT_ID"] = "ProductId";
    Attribute["FEED_VARIATION"] = "Variation";
    Attribute["FEED_SALE_START_DATE"] = "SaleStartDate";
    Attribute["FEED_SALE_END_DATE"] = "SaleEndDate";
    Attribute["FEED_SALE_PRICE"] = "SalePrice";
    Attribute["FEED_TAX_CLASS"] = "TaxClass";
    Attribute["FEED_STATUS"] = "Status";
    Attribute["FEED_CATEGORIES"] = "Categories";
    Attribute["FEED_PARENT_SKU"] = "ParentSku";
    Attribute["FEED_QUANTITY"] = "Quantity";
    Attribute["FEED_QCSTATUS"] = "QCStatus";
    Attribute["FEED_COLOR"] = "Color";
    Attribute["FEED_BASIC_COLOR"] = "ColorBasico";
    Attribute["FEED_SIZE"] = "Size";
    Attribute["FEED_TALLA"] = "Talla";
})(Attribute || (Attribute = {}));
exports.Attribute = Attribute;
