// gsc-typescript-sdk/dist/src/Model/Category/Category.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Category = void 0;
var Category = /** @class */ (function () {
    function Category() {
        this.children = [];
    }
    Category.build = function (categoryId, name, globalIdentifier, attributeSetId) {
        var category = new Category();
        category.categoryId = categoryId;
        category.name = name;
        category.globalIdentifier = globalIdentifier;
        category.attributeSetId = attributeSetId;
        return category;
    };
    Category.fromId = function (id) {
        var category = new Category();
        category.categoryId = id;
        return category;
    };
    Category.fromName = function (name) {
        var category = new Category();
        category.name = name;
        return category;
    };
    Category.prototype.getId = function () {
        return this.categoryId;
    };
    Category.prototype.getName = function () {
        return this.name;
    };
    Category.prototype.getGlobalIdentifier = function () {
        return this.globalIdentifier;
    };
    Category.prototype.getAttributeSetId = function () {
        return this.attributeSetId;
    };
    Category.prototype.addChild = function (category) {
        this.children.push(category);
    };
    Category.prototype.getChildren = function () {
        return this.children;
    };
    return Category;
}());
exports.Category = Category;
