// gsc-typescript-sdk/dist/src/Model/Feed/Feed.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Feed = void 0;
var FeedErrors_1 = require("./FeedErrors");
var FeedWarnings_1 = require("./FeedWarnings");
var Feed = /** @class */ (function () {
    function Feed(id, status, action, creation, updated, source, totalRecords, processedRecords, failedRecords, errors, warnings, failureReports) {
        this.id = id;
        this.status = status;
        this.action = action;
        this.source = source;
        this.totalRecords = totalRecords;
        this.processedRecords = processedRecords;
        this.failedRecords = failedRecords;
        this.errors = errors;
        this.warnings = warnings;
        this.failureReports = failureReports;
        this.creation = creation ? new Date(creation) : null;
        this.updated = updated ? new Date(updated) : null;
        if (!this.errors)
            this.errors = new FeedErrors_1.FeedErrors();
        if (!this.warnings)
            this.warnings = new FeedWarnings_1.FeedWarnings();
    }
    Feed.prototype.getId = function () {
        return this.id;
    };
    Feed.prototype.getStatus = function () {
        return this.status;
    };
    Feed.prototype.getSource = function () {
        return this.source;
    };
    Feed.prototype.getAction = function () {
        return this.action;
    };
    Feed.prototype.getCreationDate = function () {
        return this.creation;
    };
    Feed.prototype.getUpdatedDate = function () {
        return this.updated;
    };
    Feed.prototype.getTotalRecords = function () {
        return this.totalRecords;
    };
    Feed.prototype.getProcessedRecords = function () {
        return this.processedRecords;
    };
    Feed.prototype.getFailedRecords = function () {
        return this.failedRecords;
    };
    Feed.prototype.getErrors = function () {
        return this.errors;
    };
    Feed.prototype.getWarnings = function () {
        return this.warnings;
    };
    Feed.prototype.getFailureReports = function () {
        return this.failureReports;
    };
    return Feed;
}());
exports.Feed = Feed;
