// gsc-typescript-sdk/dist/src/Model/QualityControl/QualityControl.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.QualityControl = void 0;
var EmptyArgumentException_1 = require("../../Exception/EmptyArgumentException");
var QualityControl = /** @class */ (function () {
    function QualityControl(sellerSku, status, dataChanged, reason) {
        if (dataChanged === void 0) { dataChanged = null; }
        if (reason === void 0) { reason = null; }
        this.sellerSku = sellerSku;
        this.status = status;
        this.dataChanged = dataChanged;
        this.reason = reason;
        if (!sellerSku)
            throw new EmptyArgumentException_1.EmptyArgumentException("SellerSku");
        if (!status)
            throw new EmptyArgumentException_1.EmptyArgumentException("Status");
    }
    QualityControl.prototype.getSellerSku = function () {
        return this.sellerSku;
    };
    QualityControl.prototype.getStatus = function () {
        return this.status;
    };
    QualityControl.prototype.getDataChanged = function () {
        return this.dataChanged;
    };
    QualityControl.prototype.getReason = function () {
        return this.reason;
    };
    return QualityControl;
}());
exports.QualityControl = QualityControl;
