// gsc-typescript-sdk/dist/src/Model/QualityControl/QualityControls.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.QualityControls = void 0;
var QualityControls = /** @class */ (function () {
    function QualityControls() {
        this.collection = [];
    }
    QualityControls.prototype.findBySellerSku = function (SellerSku) {
        var status = this.collection.find(function (status) { return status.getSellerSku() === SellerSku; });
        if (!status)
            return null;
        return status;
    };
    QualityControls.prototype.searchByStatus = function (status) {
        var result = [];
        this.collection.forEach(function (qualityControl) {
            if (qualityControl.getStatus() == status) {
                result.push(qualityControl);
            }
        });
        return result;
    };
    QualityControls.prototype.all = function () {
        return this.collection;
    };
    QualityControls.prototype.add = function (qualityControl) {
        this.collection.push(qualityControl);
    };
    return QualityControls;
}());
exports.QualityControls = QualityControls;
