// gsc-typescript-sdk/dist/src/Model/Order/Adress.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Address = void 0;
var Address = /** @class */ (function () {
    function Address(firstName, lastName, phone, phone2, address, customerEmail, city, ward, region, postCode, country, address2, address3, address4, address5) {
        if (address2 === void 0) { address2 = null; }
        if (address3 === void 0) { address3 = null; }
        if (address4 === void 0) { address4 = null; }
        if (address5 === void 0) { address5 = null; }
        this.firstName = firstName;
        this.lastName = lastName;
        this.phone = phone;
        this.phone2 = phone2;
        this.address = address;
        this.customerEmail = customerEmail;
        this.city = city;
        this.ward = ward;
        this.region = region;
        this.postCode = postCode;
        this.country = country;
        this.address2 = address2;
        this.address3 = address3;
        this.address4 = address4;
        this.address5 = address5;
    }
    Address.prototype.getFirstName = function () {
        return this.firstName;
    };
    Address.prototype.getLastName = function () {
        return this.lastName;
    };
    Address.prototype.getPhone = function () {
        return this.phone;
    };
    Address.prototype.getPhone2 = function () {
        return this.phone2;
    };
    Address.prototype.getAddress = function () {
        return this.address;
    };
    Address.prototype.getAddress2 = function () {
        return this.address2;
    };
    Address.prototype.getAddress3 = function () {
        return this.address3;
    };
    Address.prototype.getAddress4 = function () {
        return this.address4;
    };
    Address.prototype.getAddress5 = function () {
        return this.address5;
    };
    Address.prototype.getCustomerEmail = function () {
        return this.customerEmail;
    };
    Address.prototype.getCity = function () {
        return this.city;
    };
    Address.prototype.getWard = function () {
        return this.ward;
    };
    Address.prototype.getRegion = function () {
        return this.region;
    };
    Address.prototype.getPostCode = function () {
        return this.postCode;
    };
    Address.prototype.getCountry = function () {
        return this.country;
    };
    return Address;
}());
exports.Address = Address;
