// gsc-typescript-sdk/dist/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Categories = exports.BusinessUnits = exports.BusinessUnit = exports.ProductData = exports.Brand = exports.Category = exports.Images = exports.Image = exports.Product = exports.Products = exports.GlobalProduct = exports.ProductStatus = exports.Configuration = exports.SellerCenterSdk = void 0;
var Configuration_1 = require("./src/Api/Configuration");
Object.defineProperty(exports, "Configuration", { enumerable: true, get: function () { return Configuration_1.Configuration; } });
var Parameters_1 = require("./src/Application/Parameters");
var HttpClient_1 = require("./src/Api/HttpClient");
var FeedManager_1 = require("./src/Services/FeedManager");
var DocumentManager_1 = require("./src/Services/DocumentManager");
var CategoryManager_1 = require("./src/Services/CategoryManager");
var BrandManager_1 = require("./src/Services/BrandManager");
var GlobalProductManager_1 = require("./src/Services/GlobalProductManager");
var WebhookManager_1 = require("./src/Services/WebhookManager");
var QualityControlManager_1 = require("./src/Services/QualityControlManager");
var ProductStatus_1 = require("./src/Contract/ProductStatus");
Object.defineProperty(exports, "ProductStatus", { enumerable: true, get: function () { return ProductStatus_1.ProductStatus; } });
var GlobalProduct_1 = require("./src/Model/Product/GlobalProduct");
Object.defineProperty(exports, "GlobalProduct", { enumerable: true, get: function () { return GlobalProduct_1.GlobalProduct; } });
var Products_1 = require("./src/Model/Product/Products");
Object.defineProperty(exports, "Products", { enumerable: true, get: function () { return Products_1.Products; } });
var Image_1 = require("./src/Model/Product/Image");
Object.defineProperty(exports, "Image", { enumerable: true, get: function () { return Image_1.Image; } });
var Images_1 = require("./src/Model/Product/Images");
Object.defineProperty(exports, "Images", { enumerable: true, get: function () { return Images_1.Images; } });
var BusinessUnit_1 = require("./src/Model/Product/BusinessUnit");
Object.defineProperty(exports, "BusinessUnit", { enumerable: true, get: function () { return BusinessUnit_1.BusinessUnit; } });
var Category_1 = require("./src/Model/Category/Category");
Object.defineProperty(exports, "Category", { enumerable: true, get: function () { return Category_1.Category; } });
var Brand_1 = require("./src/Model/Brand/Brand");
Object.defineProperty(exports, "Brand", { enumerable: true, get: function () { return Brand_1.Brand; } });
var ProductData_1 = require("./src/Model/Product/ProductData");
Object.defineProperty(exports, "ProductData", { enumerable: true, get: function () { return ProductData_1.ProductData; } });
var BusinessUnits_1 = require("./src/Model/Product/BusinessUnits");
Object.defineProperty(exports, "BusinessUnits", { enumerable: true, get: function () { return BusinessUnits_1.BusinessUnits; } });
var Categories_1 = require("./src/Model/Category/Categories");
Object.defineProperty(exports, "Categories", { enumerable: true, get: function () { return Categories_1.Categories; } });
var OrderManager_1 = require("./src/Services/OrderManager");
var ProductManager_1 = require("./src/Services/ProductManager");
var Product_1 = require("./src/Model/Product/Product");
Object.defineProperty(exports, "Product", { enumerable: true, get: function () { return Product_1.Product; } });
var SellerCenterSdk = /** @class */ (function () {
    function SellerCenterSdk(_configuration) {
        this._configuration = _configuration;
        this._parameters = Parameters_1.Parameters.fromBasics(this._configuration.userId, this._configuration.version);
        this._logger = console;
        this._httpClient = new HttpClient_1.HttpClient(this._configuration.endpoint);
    }
    Object.defineProperty(SellerCenterSdk.prototype, "config", {
        get: function () {
            return this._configuration;
        },
        enumerable: false,
        configurable: true
    });
    SellerCenterSdk.prototype.documents = function () {
        if (!(this.documentCache instanceof DocumentManager_1.DocumentManager)) {
            this.documentCache = new DocumentManager_1.DocumentManager(this._configuration, this._httpClient, this._parameters, this._logger);
        }
        return this.documentCache;
    };
    SellerCenterSdk.prototype.brands = function () {
        if (!(this.brandsCache instanceof BrandManager_1.BrandManager)) {
            this.brandsCache = new BrandManager_1.BrandManager(this._configuration, this._httpClient, this._parameters, this._logger);
        }
        return this.brandsCache;
    };
    SellerCenterSdk.prototype.categories = function () {
        if (!(this.categoryCache instanceof CategoryManager_1.CategoryManager)) {
            this.categoryCache = new CategoryManager_1.CategoryManager(this._configuration, this._httpClient, this._parameters, this._logger);
        }
        return this.categoryCache;
    };
    SellerCenterSdk.prototype.feeds = function () {
        if (!(this.feedsCache instanceof FeedManager_1.FeedManager)) {
            this.feedsCache = new FeedManager_1.FeedManager(this._configuration, this._httpClient, this._parameters, this._logger);
        }
        return this.feedsCache;
    };
    SellerCenterSdk.prototype.globalProducts = function () {
        if (!(this.globalProductCache instanceof GlobalProductManager_1.GlobalProductManager)) {
            this.globalProductCache = new GlobalProductManager_1.GlobalProductManager(this._configuration, this._httpClient, this._parameters, this._logger);
        }
        return this.globalProductCache;
    };
    SellerCenterSdk.prototype.products = function () {
        if (!(this.productCache instanceof ProductManager_1.ProductManager)) {
            this.productCache = new ProductManager_1.ProductManager(this._configuration, this._httpClient, this._parameters, this._logger);
        }
        return this.productCache;
    };
    SellerCenterSdk.prototype.webhook = function () {
        if (!(this.webhookCache instanceof WebhookManager_1.WebhookManager)) {
            this.webhookCache = new WebhookManager_1.WebhookManager(this._configuration, this._httpClient, this._parameters, this._logger);
        }
        return this.webhookCache;
    };
    SellerCenterSdk.prototype.qualityControl = function () {
        if (!(this.qualityControlCache instanceof QualityControlManager_1.QualityControlManager)) {
            this.qualityControlCache = new QualityControlManager_1.QualityControlManager(this._configuration, this._httpClient, this._parameters, this._logger);
        }
        return this.qualityControlCache;
    };
    SellerCenterSdk.prototype.orders = function () {
        if (!(this.orderCache instanceof OrderManager_1.OrderManager)) {
            this.orderCache = new OrderManager_1.OrderManager(this._configuration, this._httpClient, this._parameters, this._logger);
        }
        return this.orderCache;
    };
    return SellerCenterSdk;
}());
exports.SellerCenterSdk = SellerCenterSdk;
